cmd/tailscale/cli/ssh.go

```diff
@@ -70,12 +70,24 @@ func runSSH(ctx context.Context, args []string) error {
 		return err
 	}
 
+	prefs, err := localClient.GetPrefs(ctx)
+	if err != nil {
+		return err
+	}
+
 	// hostForSSH is the hostname we'll tell OpenSSH we're
 	// connecting to, so we have to maintain fewer entries in the
 	// known_hosts files.
 	hostForSSH := host
-	if v, ok := nodeDNSNameFromArg(st, host); ok {
-		hostForSSH = v
+	ps, ok := peerStatusFromArg(st, host)
+	if ok {
+		hostForSSH = ps.DNSName
+		if !prefs.CorpDNS {
```

Comment:
Nit: could use a comment about what we're doing here (e.g. use IP address if MagicDNS is disabled).